File: projects/EstPi.py
import random
import math
import time


def estimatePi(n):
    k = 0
    for n in range(n):
        xr = random.random() * 2
        yr = random.random() * 2
        d = math.sqrt((xr - 1.0)*(xr - 1.0) + (yr-1.0)*(yr-1.0))
        if d < 1:
            k += 1
    print("Finished in %f seconds" % (time.clock()))

    return (4 * k / n)

ePI = estimatePi(1000)
print(ePI)

File: go/EstimatePi/estpi.go
package main

import (
	"fmt"
	"math"
	"math/rand"
	"time"
)

func estimatepi(n int) float64 {
	var k float64
	startT := time.Now()
	for i := 0; i < n; i++ {
			xr := rand.Float64() * 2
			yr := rand.Float64() * 2

			dist := math.Sqrt((xr - 1.0)*(xr - 1.0) + (yr-1.0)*(yr-1.0))
			if dist < 1 {
				k++
			}
	}

	fmt.Println("Finished in", time.Now().Sub(startT).Seconds(), "seconds")

	return (4 * k / float64(n))
}

func main() {
	ePI := estimatepi(10000000)
	fmt.Println(ePI)
}
